### PR TITLE
Report physical input data size in EXPLAIN ANALYZE

### PR DESCRIPTION
**Pull request by Dith3r**
## Description

Additional information about physical data size used by connector to EXPLAIN ANALYZE in connector metrics section.

Example output.
```
     └─ TableScan[table = iceberg:part.ztest$data@1581575691215353074]
            Layout: [id:bigint, small:varchar, medium:varchar, big:varchar]
            Estimates: {rows: 483 (125.00kB), cpu: 125.00k, memory: 0B, network: 0B}
            CPU: 1.32s (62.71%), Scheduled: 3.59s (76.56%), Blocked: 0.00ns (0.00%), Output: 87 rows (1.36GB)
            Input avg.: 0.97 rows, Input std.dev.: 18.57%
            small := 2:small:varchar
            big := 4:big:varchar
            id := 1:id:bigint
            medium := 3:medium:varchar
            Physical Input: 932.06kB
```

## Non-technical explanation

Additional information about physical data size used by connector to EXPLAIN ANALYZE.

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```
# General
* Add the amount of data read from an external source during table scan to EXPLAIN ANALYZE. ({issue}`14907`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/planner/planprinter/TextRenderer.java [hunk 1]
```diff
@@ -154,23 +155,33 @@ private String printStats(PlanRepresentation plan, NodeRepresentation node)
         return output.toString();
     }
 
-    private void printMetrics(StringBuilder output, String label, Function<BasicOperatorStats, Metrics> metricsGetter, PlanNodeStats stats)
+    private void printMetrics(StringBuilder output, String label, Function<BasicOperatorStats, Metrics> metricsGetter, PlanNodeStats stats, boolean basicOperatorStats)
```
Comment (lukasz-stec) [inline core/trino-main/src/main/java/io/trino/sql/planner/planprinter/TextRenderer.java:158]: I'm not an expert here but `printMetrics` looks like a method that should only print generic `Metrics`. I wouldn't put the physical input data size here. Is there another way to print it? Maybe using something similar to `io.trino.sql.planner.planprinter.PlanNodeStats#planNodeInputDataSize`?

Comment (Dith3r) [reply core/trino-main/src/main/java/io/trino/sql/planner/planprinter/TextRenderer.java:158]: This method prints `physical read time` when `verbose` is added to explain analyze, I've assumed that physical input data should be in the same place as this information. But there is a catch—we want to print this metric without `verbose`. Extending metric to add log level looked as overkill, same with matching by name.

**Review by lukasz-stec — APPROVED** [review]
lgtm

**Comment by raunaqmorarka** [comment]
I think the approach here should be similar to https://github.com/trinodb/trino/pull/10472
Also, please add example of how the EXPLAIN output changed to the description

**Comment by lukasz-stec** [comment]
> I think the approach here should be similar to #10472

I think we want "physical input data size" displayed even if `verbose=false` and connector metrics are only displayed if `verbose=true`

**Comment by Dith3r** [comment]
@raunaqmorarka As @lukasz-stec wrote, collector metrics are displayed only for verbose output, whereas we want to display `physical input data` metric for `explain analyze` only.  Updated description.

File: core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanNodeStatsSummarizer.java [hunk 2]
```diff
@@ -216,6 +218,7 @@ else if (windowNodeStats.containsKey(planNodeId)) {
                         new Duration(planNodeBlockedMillis.get(planNodeId), MILLISECONDS),
                         planNodeInputPositions.get(planNodeId),
                         succinctBytes(planNodeInputBytes.get(planNodeId)),
+                        succinctBytes(planNodePhysicalInputDataSize.getOrDefault(planNodeId, 0L)),
```
Comment (raunaqmorarka) [inline core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanNodeStatsSummarizer.java:221]: Do we need to add it here ? Physical input data size is only about source nodes AFAIK

File: core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanNodeStatsSummarizer.java [hunk 3]
```diff
@@ -202,6 +203,7 @@ private static List<PlanNodeStats> getPlanNodeStats(TaskStats taskStats)
...
```
Comment (raunaqmorarka) [inline core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanNodeStatsSummarizer.java:206]: Do we need to add it here ? Physical input data size is only about source nodes AFAIK

File: core/trino-main/src/main/java/io/trino/sql/planner/planprinter/TextRenderer.java [hunk 4]
```diff
@@ -140,8 +140,10 @@ private String printStats(PlanRepresentation plan, NodeRepresentation node)
         if (nodeStats.getPlanNodeSpilledDataSize().toBytes() > 0) {
             output.append(format(", Spilled: %s", nodeStats.getPlanNodeSpilledDataSize()));
         }
+        if (nodeStats.getPlanNodePhysicalInputDataSize().toBytes() > 0) {
+            output.append(format(", Physical Input: %s", nodeStats.getPlanNodePhysicalInputDataSize()));
+        }
         output.append("\n");
-
```
Comment (raunaqmorarka) [inline core/trino-main/src/main/java/io/trino/sql/planner/planprinter/TextRenderer.java:144]: Unrelated change, please revert

File: core/trino-main/src/main/java/io/trino/sql/planner/planprinter/HashCollisionPlanNodeStats.java [hunk 5]
```diff
@@ -49,6 +49,7 @@ public HashCollisionPlanNodeStats(
                 planNodeBlockedTime,
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/planprinter/HashCollisionPlanNodeStats.java:49]: Please add example output to commit description

File: docs/src/main/sphinx/sql/explain-analyze.rst [hunk 6]
```diff
@@ -42,38 +42,50 @@ relevant plan nodes). Such statistics are useful when one wants to detect data a
...
+        Aggregate[type = PARTIAL, keys = [clerk], hash = [$hashvalue_2]]
+        │   Layout: [clerk:varchar(15), $hashvalue_2:bigint, count_0:bigint]
+        │   CPU: 30.00ms (13.16%), Scheduled: 30.00ms (7.19%), Blocked: 0.00ns (0.00%), Output: 1000 rows (37.11kB)
+        │   Input avg.: 818058.00 rows, Input std.dev.: 0.00%
+        │   Collisions avg.: 8247.00 (25415.23% est.), Collisions std.dev.: 0.00%
+        │   count_0 := count(*)
+        └─ ScanFilterProject[table = hive:sf1:orders, filterPredicate = ("orderdate" > DATE '1995-01-01')]
+               Layout: [clerk:varchar(15), $hashvalue_2:bigint]
+               Estimates: {rows: 1500000 (41.48MB), cpu: 35.76M, memory: 0B, network: 0B}/{rows: 816424 (22.58MB), cpu: 35.76M, memory: 0B, network: 0B}/{rows: 816424 (22.58MB), cpu: 22.58M, memory: 0B, network: 0B}
+               CPU: 180.00ms (78.95%), Scheduled: 298.00ms (71.46%), Blocked: 0.00ns (0.00%), Output: 818058 rows (12.98MB), Physical Input: 4.51MB
```
Comment (raunaqmorarka) [inline docs/src/main/sphinx/sql/explain-analyze.rst:83]: Seeing the example in docs, I can now see that this should actually be printed on the last line which prints `Input: ...`
This needs changes to `PlanPrinter#visitScanFilterAndProjectInfo` and `PlanPrinter#visitTableScan` instead of TextRenderer.

File: core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java [hunk 7]
```diff
@@ -1282,6 +1287,12 @@ private void printTableScanInfo(NodeRepresentation nodeOutput, TableScanNode nod
                             });
                 }
             }
+            PlanNodeStats nodeStats = stats.map(s -> s.get(node.getId())).orElse(null);
+            if (nodeStats != null && nodeStats.getPlanNodePhysicalInputDataSize().toBytes() > 0) {
+                nodeOutput.appendDetails(
```
Comment (raunaqmorarka) [inline core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java:1292]: printTableScanInfo is called in `visitScanFilterAndProjectInfo` as well, so this might end up double printing `Physical Input` for scanfilterproject nodes.
You can make is this change in `visitTableScan` after the call to printTableScanInfo.
It'll also be useful to include Input positions for tablescan node as it's currently missing that.

File: core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java [hunk 8]
```diff
@@ -1253,6 +1248,23 @@ private String printDynamicFilterAssignments(Map<DynamicFilterId, Symbol> filter
                     .collect(Collectors.joining(", ", "{", "}"));
         }
 
+        private void printInputInfo(NodeRepresentation nodeOutput, PlanNodeStats nodeStats)
+        {
+            if (nodeStats!= null) {
+                double filtered = 100.0d * (nodeStats.getPlanNodeInputPositions() - nodeStats.getPlanNodeOutputPositions()) / nodeStats.getPlanNodeInputPositions();
```
Comment (raunaqmorarka) [inline core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java:1254]: `filtered` is applicable only to ScanFilterProject and not TableScan, so I'd suggest not trying to make a common function out of it

**Review by raunaqmorarka — APPROVED** [review]
